recipes/nsqd.rb

```diff
@@ -25,24 +25,56 @@
 end
 
 if node['nsq']['setup_services']
-  template '/etc/init/nsqd.conf' do
-    action :create
-    source 'upstart.nsqd.conf.erb'
-    mode '0644'
-    # need to stop/start in order to reload config
-    if node['nsq']['reload_services']
-      notifies :stop, 'service[nsqd]', :immediately
-      notifies :start, 'service[nsqd]', :immediately
+
+  if node.platform_family?('debian') && node['platform_version'].to_f >= 18.04
+    template '/srv/nsqd-start.sh' do
+      action :create
+      source 'nsqd-start.sh.erb'
+      mode '0550'
+      # mode '0777'
+      owner 'nsqd'
+      group 'nsqd'
+    end
+
+    template '/etc/systemd/system/nsqd.service' do
```

Comment:
This should notify a sysctl daemon reload.

i would suggest add to the template resource

`notifies :run, 'execute[systemctl-daemon-reload]'`

and the execute block would look like
```    execute 'systemctl-daemon-reload' do
      command 'systemctl daemon-reload'
      action :nothing
```
see https://github.com/sproutsocial/infra_chef/blob/master/cookbooks/sprout_java/providers/default.rb#L199